backend/danswer/connectors/freshdesk/connector.py

```diff
@@ -0,0 +1,115 @@
+from typing import Any, List, Optional
+from bs4 import BeautifulSoup  # Add this import for HTML parsing
+from danswer.configs.app_configs import INDEX_BATCH_SIZE
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.interfaces import GenerateDocumentsOutput, PollConnector
+from danswer.connectors.models import ConnectorMissingCredentialError, Document, Section
+from danswer.utils.logger import setup_logger
+
+logger = setup_logger()
+
+
+class FreshdeskConnector(PollConnector):
+    def __init__(self, batch_size: int = INDEX_BATCH_SIZE) -> None:
+        self.batch_size = batch_size
+
+    def ticket_link(self, tid: int) -> str:
+        return f"https://{self.domain}.freshdesk.com/helpdesk/tickets/{tid}"
+
+    def build_doc_sections_from_ticket(self, ticket: dict) -> List[Section]:
+        # Use list comprehension for building sections
+        return [
+            Section(
+                link=self.ticket_link(int(ticket["id"])),
+                text=json.dumps({
+                    key: value
+                    for key, value in ticket.items()
+                    if isinstance(value, str)
+                }, default=str),
+            )
+        ]
+
+    def strip_html_tags(self, html: str) -> str:
+        soup = BeautifulSoup(html, 'html.parser')
+        return soup.get_text()
+
+    def load_credentials(self, credentials: dict[str, Any]) -> Optional[dict[str, Any]]:
+        logger.info("Loading credentials")
+        self.api_key = credentials.get("freshdesk_api_key")
+        self.domain = credentials.get("freshdesk_domain")
+        self.password = credentials.get("freshdesk_password")
+        return None
+
+    def _process_tickets(self, start: datetime, end: datetime) -> GenerateDocumentsOutput:
+        logger.info("Processing tickets")
+        if any([self.api_key, self.domain, self.password]) is None:
+            raise ConnectorMissingCredentialError("freshdesk")
+
+        freshdesk_url = f"https://{self.domain}.freshdesk.com/api/v2/tickets?include=description"
+        response = requests.get(freshdesk_url, auth=(self.api_key, self.password))
+        response.raise_for_status()  # raises exception when not a 2xx response
+
+        if response.status_code!= 204:
+            tickets = json.loads(response.content)
+            logger.info(f"Fetched {len(tickets)} tickets from Freshdesk API")
+            doc_batch: List[Document] = []
+
+            for ticket in tickets:
+                # Convert the "created_at", "updated_at", and "due_by" values to ISO 8601 strings
+                for date_field in ["created_at", "updated_at", "due_by"]:
+                    if ticket[date_field].endswith('Z'):
+                        ticket[date_field] = ticket[date_field][:-1] + '+00:00'
+                    ticket[date_field] = datetime.fromisoformat(ticket[date_field]).strftime("%Y-%m-%d %H:%M:%S")
+
+                # Convert all other values to strings
+                ticket = {
```

Comment:
make sure to put info in all the fields in the ticket and double check that when it gets converted to a document, the document looks reasonable.